### PR TITLE
add serialization support

### PR DESCRIPTION
**Pull request by xiangyan99**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by pvaneck — COMMENTED** [review]
Removing this parent `_serialization.Model` class from all these removes several public methods provided by the `_serialization.Model` class. Are we okay with this removal? Seems like this would be several breaking changes (for example, some people might be using `to_dict`, `from_dict`, etc).

Since the issue prompting this is that a user gets `KeyError: 'EntityRecognitionSkillV3'` when trying to deserialize `SearchIndexerSkillset`, could we just expose `EntityRecognitionSkillV3` in `azure.search.documents.indexes.models.__init__` so that `_serialization.Model._infer_class_models` picks it up (thus, no more key error)?

File: sdk/search/azure-search-documents/azure/search/documents/indexes/models/_models.py [hunk 1]
```diff
@@ -458,6 +422,9 @@ def _to_analyze_request(self):
             char_filters=self.char_filters,
         )
 
+    def serialize(self):
+        return self._to_analyze_request().serialize()
+
```
[the remark this answers is not in the document]
Comment (xiangyan99) [reply sdk/search/azure-search-documents/azure/search/documents/indexes/models/_models.py:427]: No. You may notice that there is no _from_analyze_request method. There is no need to deserialize the object.

**Comment by xiangyan99** [comment]
> Removing this parent `_serialization.Model` class from all these removes several public methods provided by the `_serialization.Model` class. Are we okay with this removal? Seems like this would be several breaking changes (for example, some people might be using `to_dict`, `from_dict`, etc).
> 
> Since the issue prompting this is that a user gets `KeyError: 'EntityRecognitionSkillV3'` when trying to deserialize `SearchIndexerSkillset`, could we just expose `EntityRecognitionSkillV3` in `azure.search.documents.indexes.models.__init__` so that `_serialization.Model._infer_class_models` picks it up (thus, no more key error)?

This is by design. (In fact, this is because the current implementation is wrong).

We expose a custom `EntityRecognitionSkill` model which defaults to EntityRecognitionSkillV3 with the support for EntityRecognitionSkillV1 and decided not to expose `EntityRecognitionSkillV3` directly.

File: sdk/search/azure-search-documents/azure/search/documents/indexes/models/_models.py [hunk 2]
```diff
@@ -151,6 +122,13 @@ def _from_generated(cls, skillset) -> "SearchIndexerSkillset":
         kwargs["skills"] = custom_skills
         return cls(**kwargs)
 
+    def serialize(self):
```
Comment (pvaneck) [inline sdk/search/azure-search-documents/azure/search/documents/indexes/models/_models.py:125]: Perhaps keep the same method signature for `serialize` that it was before, then propagate the arguments to the internal serialize call?

```python
def serialize(self, keep_readonly: bool = False, **kwargs: Any) -> JSON:
```

File: sdk/search/azure-search-documents/azure/search/documents/indexes/models/_models.py [hunk 3]
```diff
@@ -151,6 +122,13 @@ def _from_generated(cls, skillset) -> "SearchIndexerSkillset":
...
+        return self._to_generated().serialize()
+
+    @classmethod
+    def deserialize(cls, data):
```
Comment (pvaneck) [inline sdk/search/azure-search-documents/azure/search/documents/indexes/models/_models.py:129]: Likewise for `deserialize`?

```python
ModelType = TypeVar("ModelType", bound="SearchIndexerSkillset")

@classmethod
def deserialize(cls: Type[ModelType], data: Any, content_type: Optional[str] = None) -> ModelType:
```

**Review by pvaneck — COMMENTED** [review]
Left some comments inline. I also think we should be more detailed in the changelog regarding what public APIs are being removed as we remove the inheritance of `_serialization.Model`.

Also, this looks to be an issue relevant to this PR: https://github.com/Azure/azure-sdk-for-python/issues/30691

File: sdk/search/azure-search-documents/CHANGELOG.md [hunk 4]
```diff
@@ -6,8 +6,12 @@
 
 ### Breaking Changes
 
+- `SearchIndexerSkillset`, `SearchField`, `SearchIndex`, `AnalyzeTextOptions`, `SearchResourceEncryptionKey`, `SynonymMap`, `SearchIndexerDataSourceConnection` are not longer subclasses of `_serialization.Model`.
```
Comment (pvaneck) [inline sdk/search/azure-search-documents/CHANGELOG.md:9]:
```suggestion
- `SearchIndexerSkillset`, `SearchField`, `SearchIndex`, `AnalyzeTextOptions`, `SearchResourceEncryptionKey`, `SynonymMap`, `SearchIndexerDataSourceConnection` are no longer subclasses of `_serialization.Model`.
```

File: sdk/search/azure-search-documents/azure/search/documents/indexes/models/_models.py [hunk 5]
```diff
@@ -29,11 +28,10 @@
     SearchIndexerIndexProjections,
 )
 
-
 DELIMITER = "|"
 
 
-class SearchIndexerSkillset(_serialization.Model):
+class SearchIndexerSkillset:
```
Comment (pvaneck) [inline sdk/search/azure-search-documents/azure/search/documents/indexes/models/_models.py:34]: nit: can also remove the `super().__init__(**kwargs)` in the constructor of this class as well.

**Review by pvaneck — APPROVED** [review]

**Pull request merged**